config/index.js

```diff
@@ -0,0 +1,57 @@
+const fs = require('fs')
+const path = require('path')
+const flat = require('flat')
+
+/**
+ * Load an environment config for the current environment, which is controlled by
+ * process.env.HASHI_ENV
+ */
+function loadHashiConfigForEnvironment() {
+  const env = process.env.HASHI_ENV || 'development'
+  const envConfigPath = path.join(process.cwd(), 'config', `${env}.json`)
+
+  return getHashiConfig(envConfigPath)
+}
+
+/**
+ * Load an environment config from a specific path.
+ */
+function getHashiConfig(configPath) {
+  try {
+    const baseConfigPath = path.join(process.cwd(), 'config', `base.json`)
+    const baseConfig = JSON.parse(fs.readFileSync(baseConfigPath))
+    const envConfig = JSON.parse(fs.readFileSync(configPath))
+
+    // Load the extended config, if no extends property is defined use the base config
+    let extendsConfig = baseConfig
+    if ('extends' in envConfig && envConfig.extends !== 'base') {
+      // Determine which config this environment is extending from
+      const extendsConfigPath = path.join(
```

Comment:
Would love to hear if you specifically wanted to use the format `"extends": "filename"` in lieu of `"extends": "./filename.json"`.

Reply:
It was deliberate, as I'm not sure I would want to enable folks to think about referencing a file from elsewhere. This way it's scoped to the `/config` directory without having to add another check (though I guess someone could nefariously put `"./filename"` and the path would resolve).

Can you think of a scenario where we might want the way you've proposed? I do like that it's a bit more explicit.

Reply:
One use case I can think of is if we, in some far-flung future, wanted to reference a config that was distributed through npm. That being said, I did a little snooping and found that while files like `tsconfig.json` use file paths, others like `eslintrc` use both paths and simple names. My vote is that if we need that functionality in the future, we worry about adding it then. I was just curious as to your reasoning!

Reply:
Yeah that makes total sense! In general I try to limit functionality and only expose what's really needed, thanks for bringing this up though 🙏